install.sh

```diff
@@ -205,6 +224,78 @@ check_for_running_collectd(){
     fi
 }
 
+check_time_ntp() {
+    if which ntpdate > /dev/null; then
+        local output="$(ntpdate -q -t 2 pool.ntp.org 2> /dev/null)"
+        if [ $? -eq 0 ]; then
+            # Parse output to see what the drift is. We have to do this in awk
+            # because bash doesn't natively handle floating point.
+            if echo "$output" | awk -v max_drift=$time_sync_max_drift "{
+                if (!/step time server .+ offset/ || NF != 11)
+                    next
+
+                if ($$10 > max_drift || $$10 < -max_drift)
+                    exit 1
+                else
+                    exit 0
+            }"; then
+                echo "NTP time in sync."
+                return 0
+            else
+                echo "NTP time out of sync."
+                return 1
+            fi
+        else
+            echo "ntpdate run unsuccessful, skipping NTP time sync check."
+            return 1
+        fi
+    else
+        echo "ntpdate not found, skipping NTP time sync check."
+        return 1
+    fi
+}
+
+check_time_curl() {
+    printf "Getting current time from SignalFx ..."
+    local sfx_time=$(curl -sI $insecure https://ingest.signalfx.com | grep "^Date: " | cut -d " "  -f 2- | tr -d '\r\n')
+    check_for_err "Success"
+
+    if is_bsd; then
+        sfx_time=$(date -j -f "%a, %d %b %Y %H:%M:%S %Z" "$sfx_time" "+%s")
+    else
+        sfx_time=$(date -d "$sfx_time" +%s)
+    fi
+
+    local current_time=$(date +%s)
+    local time_diff=$(( $sfx_time - $current_time ))
+    # Take absolute value so it compares both future and present times.
+    time_diff=${time_diff#-}
+
+    if [ $time_diff -gt $time_sync_max_drift ]; then
+        echo "Local time and SignalFx time differs by $time_diff seconds (max $time_sync_max_drift)."
+        return 1
+    else
+        echo "Time is in sync with SignalFx."
```

Comment:
might not want to state this since it isn't really in sync but within a 10 second range.  do you really need to echo anything?